### PR TITLE
Fixing syscall remove

### PR DESCRIPTION
**Pull request by elbing**

### CODE REVIEW

File: lib/ap/plan9/tmpfile.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include "lib.h"
+#include "dir.h"
+
+FILE *
+tmpfile(void){
+	FILE *f;
+	static char name[]="/tmp/tmpfile_XXXXXX";
+	int n;
+
+	snprintf (name, 36, "%x", rand());
```
Comment (rminnich) [inline lib/ap/plan9/tmpfile.c]: Won't this overwrite /tmp/tmpfile_XXXXX etc. with just a random number like
1341254451
and make the file in the current directory?
Don't you want something like 
snprintf(&name[sizeof("/tmp/tmpfile_"))], 6, "%x", rand());
?

you can even do this:
#define TMPBASE "/tmp/tmpfile_"
#define XXX "XXXXXX"
static char name = TMPBASE+XXX
snprintf(&name[sizeof(TMPBASE)], len(XXX)-1, "%x", rand());

**Comment by elbing** [comment]
Trying to translate Go from Ron for tmpfile.c

**Comment by rminnich** [comment]
Is this the tmpfile you wanted? It still doesn't look right. But it's your baby.

LGTM

**Comment by elbing** [comment]
I'll replace it for anothet third party solution. Too much time wasted in it.

**Comment by elbing** [comment]
New version of tmpfile.c

File: lib/ap/plan9/tmpfile.c [hunk 2]
```diff
@@ -14,13 +14,17 @@
...
-tmpfile(void){
+tmpfile(void)
+{
 	FILE *f;
-	static char name[]="/tmp/tmpfile_XXXXXX";
+	static char name[]={TMPBASE XXX};
 	int n;
 
-	snprintf (name, 36, "%x", rand());
+	snprintf(&name[sizeof(TMPBASE)], strlen(XXX)-1, "%x", rand());
```
Comment (rminnich) [inline lib/ap/plan9/tmpfile.c]: replace strlen with sizeof

Comment (rminnich) [inline lib/ap/plan9/tmpfile.c]: Because if you don't you've got a buffer overflow.

Comment (rminnich) [inline lib/ap/plan9/tmpfile.c]: oh wait not but still ... use sizeof.

File: lib/ap/plan9/tmpfile.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#define XXX "XXXXXX"
+
+FILE *
+tmpfile(void)
+{
+	FILE *f;
+	static char name[]={TMPBASE XXX};
+	int n;
+
+	snprintf(&name[sizeof(TMPBASE)], sizeof(XXX)-1, "%x", rand());
```
Comment (dancrossnyc) [inline lib/ap/plan9/tmpfile.c]: Why not just,

    static char name[32];

    snprintf(name, sizeof(name), "/tmp/tmpfile_%x", rand());

?

File: lib/ap/plan9/tmpfile.c [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+{
+	FILE *f;
+	static char name[32];
+	int n;
+
+	snprintf(name, sizeof(name), "/tmp/tmpfile_%x", rand());
+
+	n = create(name, 64|2, 0777); /* remove-on-close */
+	if(n==-1){
+		_syserrno();
```
Comment (sevki) [inline lib/ap/plan9/tmpfile.c]: shouldn't this be something like 
```if (_plan9err == Eexist) goto FILEEXISTS;```

File: lib/ap/plan9/tmpfile.c [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+#include "lib.h"
+#include "dir.h"
+
+FILE *
+tmpfile(void)
+{
+	FILE *f;
+	static char name[32];
+	int n;
+
```
Comment (sevki) [inline lib/ap/plan9/tmpfile.c]: FILEEXISTS:

**Comment by elbing** [comment]
syserrno handles already plan9 errors.

**Pull request merged**